React/books/src/redux/Book/reducer.js

```diff
@@ -2,20 +2,23 @@ import { GET_BOOKS_FAILURE, GET_BOOKS_SUCCESS } from './actions';
 
 const initialState = {
   loading: true,
-  listBooks: []
+  listBooks: [],
+  failure: false
 };
 
 const reducer = (state = initialState, action) => {
   switch (action.type) {
     case GET_BOOKS_SUCCESS:
       return Object.assign({}, state, {
         loading: false,
-        listBooks: action.payload
+        listBooks: action.payload,
+        failure: false
       });
     case GET_BOOKS_FAILURE:
       return Object.assign({}, state, {
         loading: true,
-        listBooks: []
+        listBooks: [],
+        failure: true
```

Comment:
Why do you set loading true on failure?
Setting loading to false will save you some conditionals.